### DISCUSSION TITLE
Seeking Guidance: Achieving Consistent 100% Zoom for Mobile View in Cypress

### DISCUSSION THREAD

**Discussion by 1vivekbarnwal** [cypress-io/cypress#28319] [Questions and Help]
How can I consistently set the zoom level to 100% for a mobile device in Cypress? Presently, it fluctuates between 39% to 68%, and I've tried using cy.viewport(360, 640) and cy.get('html').invoke('css', 'zoom', '1'), but the issue persists. Any suggestions on achieving a consistent 100% zoom for a mobile view in Cypress would be greatly appreciated. 

<img width="550" alt="CypressMobileView60per" src="https://github.com/cypress-io/cypress/assets/130497406/e5a8f1f8-dc2e-48f1-829c-0696c5b505b3">

**Comment by portyu9** [comment]
Hi @1vivekbarnwal 

The `39%`–`68%` value you are seeing is most likely **Cypress Runner preview scaling**, not the zoom level of your application.

`cy.viewport(360, 640)` already sets the application viewport to `360x640`. If that viewport does not physically fit in the Cypress preview pane, Cypress automatically scales the preview so it fits. That scale can change depending on the size of the Cypress/browser window.

So this is sufficient:

```js
cy.viewport(360, 640)
```

You should **not** need:

```js
cy.get('html').invoke('css', 'zoom', '1')
```

That changes your application's CSS rendering rather than Cypress's Runner preview scale.

The important distinction is:

- `cy.viewport()` controls the actual viewport your application sees.
- The percentage displayed by Cypress is only how large that viewport is rendered inside the Runner.
- Cypress's preview scaling does not change your application's viewport calculations or responsive breakpoints.

If you want the Runner itself to display closer to `100%`, increase the size of the browser/Cypress preview pane. It is not necessary for the test itself to be accurate.

Also note that `cy.viewport()` is responsive viewport testing, not full physical mobile-device emulation.

Please mark this answer as accepted if it helped, thank you.
